jerry-core/api/jerry.c

```diff
@@ -901,6 +903,57 @@ jerry_is_feature_enabled (const jerry_feature_t feature) /**< feature to check *
           );
 } /* jerry_is_feature_enabled */
 
+/**
+ * Perform binary operation on the given operands (==, ===, <, >, etc.).
+ *
+ * @return error - if argument has an error flag or operation is unsupported
+ *         true/false - the result of the binary operation on the given operands otherwise
+ */
+jerry_value_t
+jerry_binary_operation (jerry_binary_operation_t op, /**< operation */
+                        const jerry_value_t lhs, /**< first operand */
+                        const jerry_value_t rhs) /**< second operand */
+{
+  jerry_assert_api_available ();
+
+  if (ecma_is_value_error_reference (lhs) || ecma_is_value_error_reference (rhs))
+  {
+    return jerry_throw (ecma_raise_type_error (ECMA_ERR_MSG (error_value_msg_p)));
+  }
+
+  switch (op)
+  {
+    case JERRY_BIN_OP_EQUAL:
+    {
+      return ecma_op_abstract_equality_compare (lhs, rhs);
```

Comment:
This function can throw an error [here](https://github.com/pando-project/jerryscript/blob/master/jerry-core/ecma/operations/ecma-comparison.c#L161), so I think ` jerry_return` should be used there.